content/java/concepts/array-list/terms/forEach/forEach.md

```diff
@@ -0,0 +1,112 @@
+---
+Title: '.forEach()'
+Description: 'Performs an action on each element of the ArrayList.'
+Subjects:
+  - 'Code Foundations'
+  - 'Computer Science'
+Tags:
+  - 'Arrays'
+  - 'Elements'
+  - 'Iterators'
+  - 'Loops'
+  - 'Functional Programming'
+CatalogContent:
+  - 'learn-java'
+  - 'paths/computer-science'
+---
+
+The **`.forEach()`** method performs a specified action on each element in an `ArrayList`.
+
+This method traverses each element of the `Iterable` of `ArrayList` until all the elements have been processed or an exception is raised from the action. It only iterates through the elements without modifying the ArrayList. This method does not return any value.
+
+## Syntax
+
+```pseudo
+arrayListInstance.forEach(Consumer<? super E> action);
+```
+
+This method takes an action of type `Consumer<? super E>` as an input. `Consumer<T>` is a functional interface that represents an operation which accepts a single input argument and does not return any value. As a functional interface, it can be used as the assignment target for a *lambda expression* or *method reference*.
+
+For this reason, a *lambda expression* or *method reference* is often seen to represent the action to be performed inside the `forEach()` method.

```

Comment:
```suggestion
This method takes an action of type `Consumer<? super E>` as an input. `Consumer<T>` is a functional interface representing an operation that accepts a single input argument and does not return any value. As a functional interface, it can be used as the assignment target for a lambda expression or method reference.

For this reason, a lambda expression or method reference is often seen to represent the action to be performed inside the `.forEach()` method.

```